File: Avancer_pion avec saut.js
var grille_courante=new Array ();
grille_courante[0]="B1";
grille_courante[1]="B2";
grille_courante[2]="B3";
grille_courante[3]="B4";
grille_courante[4]="0";
grille_courante[5]="0";
grille_courante[6]="N1";
grille_courante[7]="N2";
grille_courante[8]="N3";
grille_courante[9]="N4";

function avancer_pion(grille_courante,index) {
  for (let i = 0;i = 4;i++){
    let b = "B" + string(i)
    let n = "N" + string(i)

    if (grille_courante[index] == b) {

      if (grille_courante[index + 1] == "0") {
        
        let t = grille_courante[index]
        let grille_courante[index] = grille_courante[index + 1]
        let grille_courante[index + 1] = t

        return grille_courante
         }

      for (let i = 0;i = 4;i++){
        let b = "B" + string(i)
        let n = "N" + string(i)

        if (grille_courante[index + 2] == "0") {

        let t = grille_courante[index]
        let grille_courante[index] = grille_courante[index + 2]
        let grille_courante[index + 2] = t

        return grille_courante 
        } else {

          return "impossible"
        }
      } 
    } else if (grille_courante[index] == n) {

      if (grille_courante[index - 1] == "0") {
        
        let t = grille_courante[index]
        let grille_courante[index] = grille_courante[index - 1]
        let grille_courante[index - 1] = t

        return grille_courante
         } else {
            for (let i = 0;i = 4;i++){
            let b = "B" + string(i)
            let n = "N" + string(i)

              if (grille_courante[index - 1] = b / n) {
                if (grille_courante[index - 2] = "0"){

                  let t = grille_courante[index]
                  let grille_courante[index] = grille_courante[index + 2]
                  let grille_courante[index + 2] = t

                  return grille_courante
                  } else { 
                    
                    return "impossible"
                     }
              }
         }
        }
  }
if (grille_courante[index] = "0"){
  return "impossible"
}
}



let nouvelle_grille = av(grille_courante,1)

//var grille_courante = ["B1","B2","B3","B4","0","0","N1","N2","N3","N4"]
//console.log(grille_courante[0]);

File: Avancer_pion_avec_saut.py
grille_courante = ["B1", "B2", "B3", "B4", "0", "0", "N1", "N2", "N3", "N4"] #création de la grille
#B1,B2,B3,B4=pion blanc/0,0=case vide/N1,N2,N3,N4=pion noir


def avancer_pion(grille_courante, index): #appel fonction avec argument: grille et index case sélectionnée

    for i in range(1, 5): #boucle pour obtenir: B1,B2,B3,B4,N1,N2,N3,N4
      b = "B" + str(i)
      n = "N" + str(i)

      if grille_courante[index] == b: #vérifier si case = pion blanc

        if grille_courante[index + 1] == "0" :#vérifier si case suivante est vide

          t = grille_courante[index] # fonction permuter(avancer pion blanc G->D)
          grille_courante[index] = grille_courante[index + 1]
          grille_courante[index + 1] = t

          return grille_courante # retourner la grille après le mouvement
       
        for i in range(1,5): #boucle pour obtenir: B1,B2,B3,B4,N1,N2,N3,N4
          b = "B" + str(i)
          n = "N" + str(i)

          if grille_courante[index + 2] == "0" : # vérifier si la case après la case suivante le ion blanc est vide

           t = grille_courante[index] # fonction permuter(avancer pion blanc G->D)
           grille_courante[index] = grille_courante[index + 2]
           grille_courante[index + 2] = t

           return grille_courante # retourner la grille après mouvement

          else : # si la case après la case suivante le pion blanc n'est pas vide

             return "impossible" # retourner impossible
            
      elif grille_courante[index] == n: # si case sélectionnée = pion noir

         if grille_courante[index - 1] == "0" : # vérifier si case avant pion noir est vide

           t = grille_courante[index] # fonction permuter(reculer pion noir D->G)
           grille_courante[index] = grille_courante[index - 1]
           grille_courante[index - 1] = t

           return grille_courante # retourner grille après mouvement

         else : # si case avant pion noir n'est pas vide

             for i in range(1,5) : #boucle pour obtenir: B1,B2,B3,B4,N1,N2,N3,N4
               b = "B" + str(i)
               n = "N" + str(i)

               if grille_courante[index - 1] == b or n : # si pion dans case avant pion noir

                  if grille_courante[index - 2] == "0" : # si case avant case avant pion noir vide

                    t = grille_courante[index] # permuter(reculer pion noir DD->G)
                    grille_courante[index] = grille_courante[index - 2]
                    grille_courante[index - 2] = t

                    return grille_courante # retourner grille après mouvement

                  else : # si case avant case avant pion noir n'est pas vide

                     return "impossible" # retourner impossible

      if grille_courante[index] == "0": # si case sélectionnée est vide

        return "impossible" # retourner impossible


nouvelle_grille = avancer_pion(grille_courante, 0)

print(nouvelle_grille)
